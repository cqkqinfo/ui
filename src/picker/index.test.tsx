import React from 'react';
import Picker from './index';
import data from '../address-options';
import { render, fireEvent } from '@testing-library/react';
import dayjs from 'dayjs';

describe('picker选择器', () => {
  it('渲染', () => {
    const { baseElement } = render(
      <Picker cols={3} data={data}>
        显示
      </Picker>,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('城市选择', async () => {
    const { getByText, baseElement } = render(
      <Picker
        cols={3}
        data={data}
        onChange={v => {
          expect(v).toEqual(['11', '1101', '110101']);
        }}
      >
        <div>显示</div>
      </Picker>,
    );
    const children = getByText('显示');
    fireEvent.click(children);
    fireEvent.click(getByText('确定'));
    expect(baseElement).toMatchSnapshot();
  });

  it('日期选择', async () => {
    const { getByText } = render(
      <Picker mode={'date'}>
        <div>显示</div>
      </Picker>,
    );
    const children = getByText('显示');
    fireEvent.click(children);
    fireEvent.click(getByText('确定'));
    expect(getByText(dayjs().format('YYYY-MM-DD')));
  });

  it('时间选择', async () => {
    const time = dayjs().format('HH:mm');
    const { getByText } = render(
      <Picker mode={'time'}>
        <div>显示</div>
      </Picker>,
    );
    const children = getByText('显示');
    fireEvent.click(children);
    fireEvent.click(getByText('确定'));
    expect(getByText(dayjs().format(time)));
  });
});
