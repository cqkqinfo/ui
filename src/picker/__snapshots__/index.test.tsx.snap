// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`picker选择器 城市选择 1`] = `
<body
  style="overflow: hidden;"
>
  <div>
    <div>
      <div
        class="picker-children"
        extra="北京市,市辖区,东城区"
      >
        <div>
          北京市-市辖区-东城区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`picker选择器 渲染 1`] = `
<body>
  <div>
    <div>
      <div
        class="picker-children"
        extra="请选择"
      >
        显示
      </div>
    </div>
  </div>
</body>
`;
